labs/in-cluster/README.md

```diff
@@ -0,0 +1,128 @@
+# Run K6 Tests Within A Cluster
+
+## Goal
+
+This lab demonstrates running k6 within a Kubernetes cluster sending metrics to Prometheus and visualizing data in Grafana
+
+## Getting Started
+
+Switch to `labs/in-cluster` directory
+
+```bash
+cd labs/in-cluster
+```
+
+### Load Test Files
+
+Let's identify each load test file utilized for this lab.
+
+- `benchmark-k6-cluster.js`, provides a benchmark load test for cluster deployment.
+
+Now let's take a closer look at the load test file utilized for this example, and review its contents with special attention to VU code stage and different type of Checks.
+
+```bash
+# Open the file
+code scripts/benchmark-k6-cluster.js
+```
+
+## Deploying Local Cluster and ngsa-memory
+
+```bash
+# NOTE: make sure you still in `labs/in-cluster/` directory
+# create cluster, and build/deploy application, the make file used here creates a k3d cluster and deploys ngsa and istio to that cluster
+make all
+
+# after short while, check if cluster and application is deployed 
+make check
+```
+
+## Verify ngsa-memory is up and running
+
+```bash
+
+kubectl wait pods -n ngsa --all --for condition=ready --timeout=180s
+
+# check ngsa logs
+kubectl logs <ngsa-memory pod name> -n ngsa --tail 10
+```
+
+## Setup Monitoring
+
+The following instructions deploy Prometheus and Grafana in an existing cluster.
+
+The k6 has an experimental feature name [Prometheus remote write](https://k6.io/docs/results-output/real-time/prometheus-remote-write/), that can be enabled by setting the variable `K6_PROMETHEUS_RW_SERVER_URL`
+
+![Prometheus remote write configuration](images/k6-prometheus-remote-write.png)
+
+NOTE: The [k6 deployment file](labs/in-cluster/deploy/k6.yaml) has already set this configuration.
+
+### Install Grafana and Prometheus
+
+```bash
+
+#  Deploy prometheus and grafana
+#  This will also create a prometheus datasource for grafana and 2 dashboards for viewing k6 results
+kubectl apply -f deploy/monitoring
+
+# Very that prometheus and grafana pods are up an running
+kubectl wait pods -n monitoring --all --for condition=ready --timeout=180s
+
+```
+
+### Open Grafana dashboard
+
+- Enable port-forward from CLI
```

Comment:
This can probably be a separate task, but can these ports be setup like the ngsa and istio ports so that the user doesn't have to manage multiple terminals for port forwarding?

Reply:
Excellent recommendation, now Prometheus and Grafana have port forwarding implementation similar to ngsa .